ArduCopter/mode_rtl.cpp

```diff
@@ -150,7 +150,7 @@ void ModeRTL::return_start()
     }
 
     // initialise yaw to point home (maybe)
-    auto_yaw.set_mode_to_default(true);
```

Comment:
I'm pretty sure this is a regression.  ModeAuto "runs" ModeRTL by calling its init and run methods as part of its workflow.

We would previously have cross this codepath, setting "true" as being in RTL, even 'though we're not actually in RTL mode, just auto re-using RTL.

Since the new check checks the current flight mode for `is_rtl_mode()`, and that returns false for ModeAuto I think we have a change in behaviour.

I'd want to see an autotest for this case before approving this myself.

Overriding `is_rtl_mode()` (but perhaps renaming to `is_doing_RTL()`) on ModeAuto and returning true if the submode is `Submode::RTL` would probably fix the behaviour change I think I've found here.

@HefnySco are you still interested in pursuing this one?